spec/integration/api/auth_spec.rb

```diff
@@ -211,5 +212,39 @@ module Vault
         }.to_not change { subject.token }
       end
     end
+
+    describe "#aws_iam" do
+      before(:context) do
+        vault_test_client.sys.enable_auth("aws", "aws", nil)
```

Comment:
I think it would be good to also configure the header value here and then insert it, just to ensure it gets processed properly.